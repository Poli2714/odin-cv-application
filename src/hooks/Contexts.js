import { createContext } from 'react';

const IndexContext = createContext({
  activeIndex: null,
  setActiveIndex: () => {},
});

const KeyContext = createContext({
  activeKey: {
    eduActiveKey: null,
    expActiveKey: null,
  },
  setActiveKey: () => {},
});

const EduErrMsgContext = createContext({
  showEduErrMessage: false,
  setShowEduErrMessage: () => {},
});

const ExpErrMsgContext = createContext({
  showExpErrMessage: false,
  setShowExpErrMessage: () => {},
});

const PersonalInfoContext = createContext({
  personalInfo: {
    profilePicture: '',
    fullName: '',
    currentJobTitle: '',
    email: '',
    mobile: '',
    city: '',
  },
  setPersonalInfo: () => {},
});

const SummaryContext = createContext({
  summary: '',
  setSummary: () => {},
});

const ExperienceContext = createContext({
  experiences: {
    id: {
      jobTitle: '',
      employer: '',
      jobStartDate: '',
      jobEndDate: '',
      jobDescription: '',
    },
  },
  setExperiences: () => {},
});

const EducationContext = createContext({
  educations: {
    id: {
      degree: '',
      university: '',
      eduStartDate: '',
      eduEndDate: '',
      eduInfo: '',
    },
  },
  setEducations: () => {},
});

export {
  EducationContext,
  EduErrMsgContext,
  ExperienceContext,
  ExpErrMsgContext,
  IndexContext,
  KeyContext,
  PersonalInfoContext,
  SummaryContext,
};
