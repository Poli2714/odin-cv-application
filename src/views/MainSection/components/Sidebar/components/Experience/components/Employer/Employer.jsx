import { useContext } from 'react';
import PropTypes from 'prop-types';

import { InputWrapper } from 'src/components/layout';
import { Label, TextInput } from 'src/components/forms';
import { ExperienceContext } from 'src/hooks/Contexts';

function Employer({ id }) {
  const { experiences, setExperiences } = useContext(ExperienceContext);

  const handleChangeEmployer = (e, key) =>
    setExperiences({
      ...experiences,
      [key]: { ...experiences[key], employer: e.target.value },
    });

  return (
    <InputWrapper>
      <Label labelName="Employer" labelFor="employer" />
      <TextInput
        name="employer"
        onChange={e => handleChangeEmployer(e, id)}
        placeholder="e.g. Facebook"
        value={experiences[id].employer}
      />
    </InputWrapper>
  );
}

Employer.propTypes = {
  id: PropTypes.string,
};

export default Employer;
