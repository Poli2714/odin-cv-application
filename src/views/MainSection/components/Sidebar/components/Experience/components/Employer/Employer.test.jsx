import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';

import Employer from './Employer';
import { ExperienceContext } from 'src/hooks/Contexts';

vi.mock('src/components/layout', () => ({
  InputWrapper: vi.fn(({ children }) => (
    <div data-testid="input-wrapper">{children}</div>
  )),
}));

vi.mock('src/components/forms', () => ({
  Label: vi.fn(() => <label htmlFor="employer">Mock label</label>),
  TextInput: vi.fn(({ onChange, value }) => (
    <input id="employer" type="text" onChange={onChange} value={value} />
  )),
}));

function MockParenComponent({ id }) {
  const [experiences, setExperiences] = useState({
    testId: { employer: '' },
    testId2: { employer: 'hello' },
  });

  return (
    <ExperienceContext.Provider value={{ experiences, setExperiences }}>
      <Employer id={id} />
    </ExperienceContext.Provider>
  );
}

test('renders Employer', () => {
  render(<MockParenComponent id="testId" />);

  expect(screen.getByTestId('input-wrapper')).toMatchSnapshot();
});

test('renders initial employer value correctly', () => {
  render(<MockParenComponent id="testId2" />);

  expect(screen.getByLabelText('Mock label')).toHaveValue('hello');
});

test('updates employer value as a user types', async () => {
  const user = userEvent.setup();

  render(<MockParenComponent id="testId" />);
  const input = screen.getByLabelText('Mock label');

  await user.type(input, 'Google');
  expect(input).toHaveValue('Google');

  await user.clear(input);
  await user.type(input, 'Twitter');
  expect(input).toHaveValue('Twitter');
});
