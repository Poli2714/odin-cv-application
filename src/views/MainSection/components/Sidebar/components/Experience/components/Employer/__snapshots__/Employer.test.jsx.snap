// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Employer 1`] = `
<div
  data-testid="input-wrapper"
>
  <label
    for="employer"
  >
    Mock label
  </label>
  <input
    id="employer"
    type="text"
    value=""
  />
</div>
`;
